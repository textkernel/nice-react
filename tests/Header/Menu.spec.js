import React from 'react';
import HeaderMenu from '../../lib/Header/Menu';

describe('<Header.Menu />', () => {
    it('should render default header menu group', () => {
        const wrapper = shallow(<HeaderMenu>
                <HeaderMenu.Item label="Item 1" />
                <HeaderMenu.Item label="Item 2" />
        </HeaderMenu>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render alternate header menu group', () => {
        const wrapper = shallow(<HeaderMenu alternate>
                <HeaderMenu.Item label="Item 1" />
                <HeaderMenu.Item label="Item 2" />
        </HeaderMenu>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render Header Menu with custom className/style', () => {
        const wrapper = shallow(
            <HeaderMenu className="my-custom-class" style={{ top: 0 }} />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
