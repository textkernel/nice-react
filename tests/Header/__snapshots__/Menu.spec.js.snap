// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header.Menu /> should render Header Menu with custom className/style 1`] = `
<ul
  className="nice-header-menu my-custom-class"
  style={
    Object {
      "top": 0,
    }
  }
/>
`;

exports[`<Header.Menu /> should render alternate header menu group 1`] = `
<ul
  className="nice-header-menu menu-alt"
  style={Object {}}
>
  <Item
    className={null}
    context={null}
    href="#"
    key=".0"
    label="Item 1"
    level={0}
    onClick={null}
    style={Object {}}
    target={null}
    width={null}
  />
  <Item
    className={null}
    context={null}
    href="#"
    key=".1"
    label="Item 2"
    level={0}
    onClick={null}
    style={Object {}}
    target={null}
    width={null}
  />
</ul>
`;

exports[`<Header.Menu /> should render default header menu group 1`] = `
<ul
  className="nice-header-menu"
  style={Object {}}
>
  <Item
    className={null}
    context={null}
    href="#"
    key=".0"
    label="Item 1"
    level={0}
    onClick={null}
    style={Object {}}
    target={null}
    width={null}
  />
  <Item
    className={null}
    context={null}
    href="#"
    key=".1"
    label="Item 2"
    level={0}
    onClick={null}
    style={Object {}}
    target={null}
    width={null}
  />
</ul>
`;
